### PR TITLE
Close subscription, when refreshing it

### PR DESCRIPTION
**Pull request by Gafurov0ivan**
All connected clients in redis stay active, even subscription already refreshed.
Need to close subscription.

### CODE REVIEW

File: src/main/java/com/github/fred84/queue/DelayedEventService.java [hunk 1]
```diff
@@ -225,6 +225,7 @@ private DelayedEventService(Builder builder) {
     }
 
     void refreshSubscriptions() {
+        subscriptions.forEach((key, value) -> value.subscription.dispose());
```
Comment (fred84) [inline src/main/java/com/github/fred84/queue/DelayedEventService.java]: Let's call it inside lamda passed to `replaceAll`

```
subscriptions.replaceAll((k, v) -> {
  v.subscription.dispose();
  createNewSubscription(v);
});

File: src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java [hunk 2]
```diff
@@ -558,6 +559,38 @@ void failedEventsDuringRefreshWouldBeHandledLater() throws InterruptedException
                 .collect(toSet());
 
         assertThat(leftForProcessing, equalTo(unhandledIds));
+        eventService.refreshSubscriptions();
```
Comment (fred84) [inline src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java]: Why you are calling refresh at this end of this test?

File: src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java [hunk 3]
```diff
@@ -558,6 +559,38 @@ void failedEventsDuringRefreshWouldBeHandledLater() throws InterruptedException
...
+        eventService.refreshSubscriptions();
+        eventService.refreshSubscriptions();
+    }
+
+    @Test
+    void closeClientsAfterRefresh()  {
+        enqueue(10);
```
Comment (fred84) [inline src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java]: Why do you need all this latches and enqueed elements here? It should be enough just to:

- subscribe
- count clients
- refresh multiple times
- count clients

File: docker-compose.yml [hunk 4]
```diff
@@ -5,3 +5,10 @@ services:
     image: redis:4-alpine
     ports:
      - "6379:6379"
+     
+# Use only in linux
+#  toxiproxy:
+#         image: shopify/toxiproxy
```
Comment (Svetopolk) [inline docker-compose.yml]: why is it commented?
mac user can do "docker-compose up redis"
instead docker-compose up

**Review by fred84 — CHANGES_REQUESTED** [review]

File: src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java [hunk 5]
```diff
@@ -405,9 +405,12 @@ void blockingSubscriberErrorHandling() throws InterruptedException {
                     latch.countDown();
 
                     switch ((Integer.parseInt(e.getId())) % 4) {
-                        case 0: throw new RuntimeException("no-no");
-                        case 1: return true;
-                        default: return false;
+                        case 0:
```
Comment (fred84) [inline src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java]: This changes is not related to subscriptions

File: src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java [hunk 6]
```diff
@@ -565,32 +574,18 @@ void failedEventsDuringRefreshWouldBeHandledLater() throws InterruptedException
...
-                    } catch (InterruptedException ex) {
-                        throw new RuntimeException(ex);
-                    }
-                    return true;
-                },
-                1
-        );
-        
-        assertConnectedClients(4);
+        int initNumber = countActiveClients();
```
Comment (fred84) [inline src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java]: Let's call it `serviceConnectionsCount`

File: src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java [hunk 7]
```diff
@@ -684,7 +679,7 @@ private void sleepMillis(long duration) {
         }
     }
 
-    private void assertConnectedClients(int expected) {
-        assertThat(connection.clientList().split("\\r?\\n").length, is(expected));
+    private int countActiveClients() {
```
Comment (fred84) [inline src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java]: Let's rename it to `getNumberOfActiveClients`

File: docker-compose.yml [hunk 8]
```diff
@@ -4,11 +4,11 @@ services:
...
-     
-# Use only in linux
-#  toxiproxy:
-#         image: shopify/toxiproxy
-#         ports:
-#           - "8474:8474"
-#         network_mode: "host"
+      - "6379:6379"
+
+  # works fine in Ubuntu, Mac is questionable
```
Comment (fred84) [inline docker-compose.yml]: Let's make it clear: "Works only on linux"

File: src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java [hunk 9]
```diff
@@ -560,6 +561,21 @@ void failedEventsDuringRefreshWouldBeHandledLater() throws InterruptedException
         assertThat(leftForProcessing, equalTo(unhandledIds));
     }
 
+    @Test
+    void closeClientsAfterRefresh() {
+        enqueue(10);
```
Comment (fred84) [inline src/test/java/com/github/fred84/queue/DelayedEventServiceTest.java]: Why do we need to enqueue elements?

**Review by fred84 — APPROVED** [review]
